tests/integrations/google_ai_studio/google_ai_studio_test.py

```diff
@@ -0,0 +1,107 @@
+import asyncio
+import os
+from typing import Any, Optional
+
+import pytest
+
+from weave.trace.weave_client import WeaveClient
+from weave.trace_server import trace_server_interface as tsi
+
+
+def flatten_calls(
+    calls: list[tsi.CallSchema], parent_id: Optional[str] = None, depth: int = 0
+) -> list:
+    def children_of_parent_id(id: Optional[str]) -> list[tsi.CallSchema]:
+        return [call for call in calls if call.parent_id == id]
+
+    children = children_of_parent_id(parent_id)
+    res = []
+    for child in children:
+        res.append((child, depth))
+        res.extend(flatten_calls(calls, child.id, depth + 1))
+
+    return res
```

Comment:
You shouldn't need these lower-level interfaces for tests.  See other integration tests for examples:
https://github.com/wandb/weave/blob/master/tests/integrations/anthropic/anthropic_test.py